### PR TITLE
Integrate title suggestion in attribute picker

### LINKED ISSUE #101: Integrate addressViewModel into Attribute pickerScreen

### PR DESCRIPTION
**Pull request by Armetiis**
Closes #101

This PR integrates feature implemented in PR #84 into the Attribute Picker Screen.
Additional changes are trivial changes (e.g. initialize AddressViewModel and add it to navgraph)

This is what kind of title might be suggested:
image

No additional tests (feature is already tested and screen too. This PR doesn't anything that would justify new tests)

### CODE REVIEW

**Review by sean-prz — CHANGES_REQUESTED** [review]
Nice work.

We should try to decrease the number of calls to the API, currently we call it 6 times for the same address.

Also when the title is too long it overflows the topBar, we can either make the fontsize smaller or truncate the title, as you want. 

image


Address these two things and we're good to go

**Comment by Armetiis** [comment]
> We should try to decrease the number of calls to the API, currently we call it 6 times for the same address.

I've tried to wrap the call to search in a LaunchedEffect. I tried different implementations but whatever I did I either did not succeed to reduce the number of calls, either failed to make it work (search returned no result). Any suggestions?

File: app/src/main/java/com/github/se/cyrcle/ui/addParking/attributes/AttributesPicker.kt [hunk 1]
```diff
@@ -64,19 +66,21 @@ import com.mapbox.maps.plugin.gestures.generated.GesturesSettings
...
-    mapViewModel: MapViewModel
+    mapViewModel: MapViewModel,
+    addressViewModel: AddressViewModel
 ) {
-  val location = mapViewModel.selectedLocation.collectAsState().value!!
-  val title = remember {
-    mutableStateOf(
-        "${("%.2f").format(location.center.longitude())}, ${("%.3f").format(location.center.latitude())}")
-  }
+  val title = remember { mutableStateOf("") }
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/addParking/attributes/AttributesPicker.kt:72]: Out of curiosity, why was this changed?

**Review by LouisJamesVasseur — APPROVED** [review]
Very cool feature! I find it very well implemented both from a UI and a code perspective! Great job!

File: app/src/main/java/com/github/se/cyrcle/ui/addParking/attributes/AttributesPicker.kt (hunk 1 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/addParking/attributes/AttributesPicker.kt:72]: The initial value of the title is irrelevant, given that as soon as the titleSuggestion is loaded from the AddressViewModel, it will get replaced. The previous value was just a placeholder @sean-prz used to avoid empty titled parkings and facilitate debugging

**Review by sean-prz — APPROVED** [review]
Perfect, thanks for the changes
